distutils/command/build_ext.py

```diff
@@ -217,14 +217,18 @@ def finalize_options(self):  # noqa: C901
                 new_lib = os.path.join(new_lib, suffix)
             self.library_dirs.append(new_lib)
 
-        # For extensions under Cygwin, Python's library directory must be
+        # For extensions under Cygwin and MinGW, Python's library directory must be
         # appended to library_dirs
-        if sys.platform[:6] == 'cygwin':
+        if sys.platform[:6] == 'cygwin' or is_mingw():
             if not sysconfig.python_build:
+                config_dir_name = os.path.basename(sysconfig.get_config_var('LIBPL'))
```

Comment:
Is this still needed with the recent relocation fix i.e. could we use LIBPL as is?

Reply:
Yeah, it'll work for mingw, but I don't know whether it would work for Cygwin.

Reply:
On cygwin it returns `'/usr/lib/python3.9/config-3.9-x86_64-cygwin'` and `'/usr/lib/python3.9/config'` doesn't exist. tbh I don't quite understand what this is doing in the first place..

Reply:
I'm also not sure, I copied it from the old patches. Maybe we don't need it at all?

Reply:
Without having tested it, only the build_python case makes somewhat sense, but since this version of distutils isn't used for building CPython internal extensions https://github.com/pypa/setuptools/commit/336facde9301e041b2e33a7b46fe1d37991a7ed3 I think it might not be needed at all, yes.

Reply:
> Without having tested it, only the build_python case makes somewhat sense, but since this version of distutils isn't used for building CPython internal extensions https://github.com/pypa/setuptools/commit/336facde9301e041b2e33a7b46fe1d37991a7ed3 I think it might not be needed at all, yes.

I removed this part of the patch and things seems to be working, so I guess it's all fine.